core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1330,7 +1330,7 @@ RexNode simplifyAnd(RexCall e, RexUnknownAs unknownAs) {
 
     final SargCollector sargCollector = new SargCollector(rexBuilder, true);
     operands.forEach(t -> sargCollector.accept(t, terms));
-    if (sargCollector.map.values().stream().anyMatch(b -> b.complexity() > 1)) {
+    if (sargCollector.needToFix(terms.size())) {
```

Comment:
What if we always convert to sarg format? Are there drawbacks?

Reply:
The `drawback` is there are many unnecessary plan change.

Reply:
What do you mean `unnecessary plan change`?
Do you mean plan changes when compared with 1.25 or 1.26?

I guess 1.26 is not really viable (since there are significant issues with Sarg), so I would skip 1.26 from consideration with regard to plan changes.

Reply:
For example, a simple `$0=1` would be converted to `Sarg` which is meaningless.